modules/grafana-server/main.tf

```diff
@@ -0,0 +1,90 @@
+########################################################################################################################################################################################################################
+#
+# This template deploys an ec2-instance configured as a grafana server which uses prometheus as a data source.
+# It also stands up and configures an RDS instance to hold the data.
+#
+########################################################################################################################################################################################################################
+
+# ---------------------------------------------------------------------------------------------------------------------
+# THESE TEMPLATES REQUIRE TERRAFORM VERSION 0.8 AND ABOVE
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.9.3"
+}
+
+resource "aws_instance" "grafana_instance" {
+  instance_type         = "${var.instance_type}"
+  ami                   = "${var.ami_id}"
+  key_name              = "${var.ssh_key_name}"
+  subnet_id             = "${var.subnet_id}"
+  associate_public_ip_address = "${var.associate_public_ip_address}"
+  vpc_security_group_ids = ["${aws_security_group.grafana_security_group.id}", "${var.aws_security_group_ids}"]
+  user_data              = "${var.user_data == "" ? data.template_file.grafana_user_data.rendered : var.user_data}"
+  tags = "${merge(var.tags, map("Name", "${var.instance_name}"))}"
+
+  # Copies all files under the dashboards dir to the instance's /usr/share/grafana/public/dashboards directory,
+  #    so we can dynamically load dashboards from source.
+  provisioner "file" {
```

Comment:
So why the file provisioner instead of just including the dashboard files on the AMI via Packer?

Reply:
I wanted to be able to change the dashboards on every terraform run, instead of creating new AMI every time we wanted to add a dashboard.